packages/create-llama/templates/components/ui/shadcn/chat/chat-messages.tsx

```diff
@@ -23,6 +24,12 @@ export default function ChatMessages(
   const showReload =
     props.reload && !props.isLoading && isLastMessageFromAssistant;
   const showStop = props.stop && props.isLoading;
+  const content = lastMessage?.content ?? "";
+  const deferredContent = useDeferredValue(content);
+  const isPending = props.isLoading ? !isLastMessageFromAssistant : (
+    props.isLoading &&
+    isLastMessageFromAssistant && content === deferredContent
+  );
```

Comment:
@himself65, this condition is a bit hard to read so a comment could help. E.g., Why do we have `props.isLoading` again in the second clause (when `props.isLoading` is `false`).

Reply:
I thought twice and found that I should have made the logic better. updated